medicines/doc-index-updater/src/service_bus_client/mod.rs

```diff
@@ -84,10 +103,26 @@ impl DocIndexUpdaterQueue {
                 }
             })?;
 
-        Ok(message.parse::<T>().map_err(|_| {
-            tracing::warn!("Message found could not be parsed ({:?})", message);
-            RetrieveFromQueueError::ParseError(message)
-        })?)
+        let body = peek_lock.body();
+
+        match body.parse::<T>() {
+            Ok(message) => {
+                tracing::info!(
+                    "Message found perfectly parseable ({:?}).\n{:?}",
+                    body,
+                    message.to_json_string()
+                );
+                Ok(RetrievedMessage { message, peek_lock })
+            }
+            Err(_) => {
+                tracing::error!(
+                    "Message found could not be parsed ({:?}). Gonna go ahead and delete it.",
+                    body
+                );
+                let _ = peek_lock.delete_message().await;
```

Comment:
How difficult would it be to set up some bad message store? Just deleting them feels a bit flippant.

Reply:
Yeah, spoke to @StuartHarris about this. Creating another queue would be easy: a copy-paste in the terraform. However, we now have the `DocIndexUpdaterQueue`, which handles both the sending and receiving, keeping them compatible. Also, we'd have to clean up `"bad-message-queue"`, and what would be the plan for that? Is it an extra feature we require, when we control the shape of the sending and the receiving?

Reply:
Fair enough. I'll feel a lot more comfortable about this once we have tests across all of this.